jgrapht-core/src/main/java/org/jgrapht/alg/matching/blossom/v5/Edge.java

```diff
@@ -0,0 +1,173 @@
+/*
+ * (C) Copyright 2018-2018, by Timofey Chudakov and Contributors.
+ *
+ * JGraphT : a free Java graph-theory library
+ *
+ * This program and the accompanying materials are dual-licensed under
+ * either
+ *
+ * (a) the terms of the GNU Lesser General Public License version 2.1
+ * as published by the Free Software Foundation, or (at your option) any
+ * later version.
+ *
+ * or (per the licensee's choosing)
+ *
+ * (b) the terms of the Eclipse Public License v1.0 as published by
+ * the Eclipse Foundation.
+ */
+package org.jgrapht.alg.matching.blossom.v5;
+
+import org.jgrapht.util.FibonacciHeapNode;
+
+/**
+ * This class is a supporting data structure for Kolmogorov's Blossom V algorithm.
+ * <p>
+ * Represents an edge between two nodes. Also, the minimum weight perfect matching problem is formulated on
+ * an undirected graph, each edge has direction, i.e. it is an arc. According to this direction it is present in
+ * two circular doubly linked lists of incident edges. The references to the next and previous edges of this list
+ * are maintained via {@link Edge#next} and {@link Edge#prev} arrays.
+ * <p>
+ * For example, let a $e = \{u, v\}$ be an edge in the graph $G = (V, E)$. Let's assume that after initialization
+ * this edge has become directed from $u$ to $v$, i.e. now $e = (u, v)$. Then now edge $e$ belongs to the linked lists
+ * {@code u.first[0]} and {@code v.first[1]}. In other words, $e$ is an outgoing edge of $u$ and an incoming edge
+ * of $v$. For convenience during computation, {@code e.head[0] = v} and {@code e.head[1] = u}. Therefore, while
+ * iterating over incident edges of a node {@code x} in the direction {@code dir}, we can easily access opposite node
+ * by {@code x.head[dir]}.
+ *
+ * @author Timofey Chudakov
+ * @see KolmogorovMinimumWeightPerfectMatching
+ * @since June 2018
+ */
+class Edge {
```

Comment:
Even though everything is now properly namespaced in the blossom.v5 package, it still might be a little confusing to have classes with very generic names such as Edge.  The reason is that in the Javadoc, there's a flat list of all top-level classes:

https://jgrapht.org/javadoc/allclasses-frame.html

(This is the source for one of the frames on the main Javadoc page.)

So someone browsing that may be misled into thinking that this is a general-purpose Edge class (until they click through to read the details).

So some short prefix for such class names is probably warranted, maybe e.g. BlossomVEdge, BlossomVTree, etc like you already have for BlossomVDebugger?